src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1521,6 +1615,81 @@ public bool MoveNext()
             return true;
         }
     }
+
+    /// <summary>
+    /// The simple enumerator class is used for the range operator '..'
+    /// in expressions like 'A'..'B' | ForEach-Object { $_ }
+    /// </summary>
+    internal class CharRangeEnumerator : IEnumerator
```

Comment:
The code is basically the same as RangeEnumerator, I wonder if there is a clever way we could avoid the duplication.

One thought - you could maybe derive from RangeEnumerator and just override Current.

It's probably not worth spending time on this though - C# doesn't handle this scenario nicely like C++ template would.

Reply:
I started with this but it looked too confusing. In addition, I thought that if we wanted to enhance this operator in the future, it would be easier if each type would have its own enumerator. Also possible we can make optimization and reduce string allocations.